### PR TITLE
Adds DEVELOPMENT.md

### LINKED ISSUE #26: Add DEVELOPMENT.md
It would be good for the repo to have a document to help new contributors to ramp up for example how to setup their development workstation, any tools and requirements.
It would be useful to align with the Go version, for example serving requires Go 1.12 (released on Feb 25 2019)

Doc cloud include how to build, configure and run the CLI.
Running tests and maybe a TDD approach
References to doc/tutorials on libraries being leverage like knative/pkg or k8s client-go

Here are the DEVELOPMENT.md files in other repos as examples:
https://github.com/knative/serving/blob/master/DEVELOPMENT.md
https://github.com/knative/eventing/blob/master/DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes #26 

Also adds CONTRIBUTING.md which points to knative/docs and links the development guide from README.md.

### CODE REVIEW

**Comment by navidshaikh** [comment]
Does this PR need more content to be added? Looking forward to your review @sixolet @evankanderson @cppforlife @cdrage

**Comment by cdrage** [comment]
This LGTM :+1:

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/): For
+   managing development environments
+
+### Create a cluster
+
+1. [Set up a Knative](https://github.com/knative/docs/blob/master/install/README.md#install-guides)
+   - You can also setup using limited install guides for eg: Minikube/Minishift.
+
+### Setup your environment
```
Comment (sixolet) [inline DEVELOPMENT.md]: This is not actually required for knative client development, since we're using the new modules support in golang.

**Comment by bbrowning** [comment]
@navidshaikh Do you plan to address the feedback from @sixolet about environment setup and golang modules in this PR? Or in a subsequent one?

File: DEVELOPMENT.md (hunk 1 above)
Comment (navidshaikh) [reply DEVELOPMENT.md]: Updated DEVELOPMENT.md and added a notes at the bottom if one is using `GOPATH`.

**Comment by navidshaikh** [comment]
@bbrowning : Updated the PR addressing the review comments.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+   managing development environments
+
+### Create a cluster
+
+1. [Set up a Knative](https://www.knative.dev/docs/install/)
+   - You can also setup using limited install guides for eg: Minikube/Minishift.
+
+### Setup your environment
+
+Let's setup local development envrionment.
```
Comment (bbrowning) [inline DEVELOPMENT.md]: Is there more that should be in this section? If it's just the one sentence, perhaps we don't need this sentence or heading at all?

Comment (navidshaikh) [reply DEVELOPMENT.md]: Right.. removed a heading and sentence altogether.

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+It builds `kn` binary in your current directory. You can start playing with it.
+
+**Notes:**
+
+- For building, Go `1.11.4` is required [due to a go mod issue](https://github.com/golang/go/issues/27925).
+- If you are building in your `$GOPATH` folder, you need to specify `GO111MODULE` for building it
+
+```sh
+# if you are building in your $GOPATH
+GO111MODULE=on go build ./cmd/...
```
[the remark this answers is not in the document]
Comment (navidshaikh) [reply DEVELOPMENT.md]: `GOPATH` mention in earlier step was a left over in recent update to the PR.
I removed the `$GOPATH` reference in earlier step. Thanks for pointing this out!

**Pull request merged**